pcdsdevices/signal.py

```diff
@@ -335,21 +341,56 @@ class UnitConversionDerivedSignal(DerivedSignal):
         Keyword arguments are passed to the superclass.
     """
 
+    derived_units: str
+    original_units: str
+
     def __init__(self, derived_from, *,
                  derived_units: str,
                  original_units: typing.Optional[str] = None,
+                 user_offset: typing.Optional[typing.Any] = None,
                  **kwargs):
         self.derived_units = derived_units
         self.original_units = original_units
+        self._user_offset = user_offset
         super().__init__(derived_from, **kwargs)
 
     def forward(self, value):
         '''Compute derived signal value -> original signal value'''
+        if self.user_offset is not None:
+            value = value + self.user_offset
         return convert_unit(value, self.derived_units, self.original_units)
 
     def inverse(self, value):
         '''Compute original signal value -> derived signal value'''
-        return convert_unit(value, self.original_units, self.derived_units)
+        derived_value = convert_unit(value, self.original_units,
+                                     self.derived_units)
+        if self.user_offset is not None:
+            derived_value = derived_value - self.user_offset
+        return derived_value
+
+    @property
+    def user_offset(self) -> typing.Optional[typing.Any]:
+        """A user-specified offset in *derived*, user-facing units."""
+        return self._user_offset
+
+    @user_offset.setter
+    def user_offset(self, offset):
+        self._user_offset = offset
+        self._recalculate_position()
+
```

Comment:
I suspect a very quick request once a user gets their hands on this will be to ask for a method that updates the offset such that the current position matches set_position, e.g. lxt.set_position(0) could be used to zero the position value.

I'm not sure what to call this method or whether it is worth doing before someone asks.

Reply:
I think that's a "calibrate"-style operation. Reasonable name `calibrate_zero_position` or just `set_zero_position`?

Reply:
OK, will add.